v-next/hardhat/src/internal/core/tasks/task-manager.ts

```diff
@@ -219,6 +219,35 @@ export class TaskManagerImplementation implements TaskManager {
         );
       }
     });
+
+    const globalOptionDefinitionsByShortName: GlobalOptionDefinitions =
+      new Map();
+    for (const globalOptionEntry of globalOptionDefinitions.values()) {
+      if (globalOptionEntry.option.shortName !== undefined) {
+        globalOptionDefinitionsByShortName.set(
+          globalOptionEntry.option.shortName,
+          globalOptionEntry,
+        );
+      }
+    }
+    const optionShortNames = Object.values(taskDefinition.options)
+      .map((option) => option.shortName)
+      .filter((shortName) => shortName !== undefined);
+
+    optionShortNames.forEach((argName) => {
+      const globalOptionEntry = globalOptionDefinitions.get(argName);
```

Comment:
I think this is incorrect, because `argName` here is a short name, but the keys of `globalOptionDefinitions` are long names. I verified this with this example config:

```ts
const fooTask = task("foo")
  .addFlag({
    name: "fooOption",
    shortName: "m",
  })
  .setAction(() => {}).build()

const myPlugin: HardhatPlugin = {
  id: "my-plugin",
  globalOptions: [{
    name: "myPluginOption",
    shortName: "m",
    type: ArgumentType.BOOLEAN,
    description: "An example plugin option",
    defaultValue: false,
  }]
}

const config: HardhatUserConfig = {
  plugins: [myPlugin],
  tasks: [fooTask],
```

This should've raised an error, but didn't.